feat: add copy selection as context feature

yinyu985 opened the pull request:
- Implement copySelection function with precise line numbers
- Add support for file paths and workspace-relative paths
- Include comprehensive unit tests for copy-selection functionality
- Register 'Copy Selection as Context' command in editor context menu
- Maintain copy-context's existing markdown formatting style
- Add telemetry tracking for user analytics

Closes #10

Fralleee commented on package.json:
```diff
@@ -205,5 +216,5 @@
 		"watch": "npm run esbuild-base -- --watch"
 	},
-	"version": "0.8.0"
+	"version": "0.8.3"
```
No need to set the version; this is set by release CI.

Fralleee reviewed (changes requested):
Hey @yinyu985,

Great feature proposal!
Could you please address the conflicts and suggestions I added?

Then, I'll be happy to merge this. 👍

yinyu985 committed:
feat: add copy selection as context feature

yinyu985 committed:
fix: address reviewer feedback on package.json

yinyu985 commented:
Hi @Fralleee 

Thanks for the detailed feedback! I've addressed all your suggestions:

✅ **Version**: Reverted to 0.8.0 as you noted that version is managed by release CI

✅ **Menu configuration**: Fixed the menu placement - tab copy commands remain in `editor/title/context`, and the new `Copy Selection` command is only in `editor/context` (when text is selected)

✅ **Command title**: Removed the category and simplified the title to just "Copy Selection"

All functionality is working correctly and the PR should now be ready for merge. Thanks again for your guidance on this! 👍
